nncf/torch/pruning/filter_pruning/algo.py

```diff
@@ -253,12 +256,38 @@ def _init_pruned_modules_params(self):
             self.pruning_quotas[cluster.id] = np.floor(self._modules_out_channels[cluster.elements[0].node_name] \
                                                        * self.pruning_quota)
 
+    def _calculate_output_shape(self, graph: NNCFGraph, node: NNCFNode) -> Tuple[int, ...]:
+        """
+        Calculates output shape of convolution layer by input edge.
+
+        :param graph: the model graph
+        :param node: node from NNCF graph
+        :return: output shape
+        """
+        in_edge = graph.get_input_edges(node)[0]
+        shape = list(in_edge.tensor_shape)[2:]
+        attrs = node.layer_attributes
+
+        assert isinstance(attrs, ConvolutionLayerAttributes)
+
+        for i, _ in enumerate(shape):
+            if attrs.transpose:
+                shape[i] = (shape[i] - 1) * attrs.stride[i] - 2 * attrs.padding_values[i] + attrs.kernel_size[i]
+            else:
+                shape[i] = (shape[i] + 2 * attrs.padding_values[i] - attrs.kernel_size[i]) // attrs.stride[i] + 1
+        return tuple(shape)
+
     def flops_count_init(self) -> None:
         graph = self._model.get_original_graph()
         for node in graph.get_nodes_by_types([v.op_func_name for v in NNCF_GENERAL_CONV_MODULES_DICT]):
-            out_edge = graph.get_output_edges(node)[0]
-            out_shape = out_edge.tensor_shape
-            self._modules_out_shapes[node.node_name] = out_shape[2:]
+            if graph.get_output_edges(node):
+                out_edge = graph.get_output_edges(node)[0]
+                out_shape = out_edge.tensor_shape[2:]
+            else:
+                # For disconnected NNCFGraph when convolution layers have no output edge
+                out_shape = self._calculate_output_shape(graph, node)
+                nncf_logger.error("Node %s have no output edge in NNCFGraph", node.node_name)
+            self._modules_out_shapes[node.node_name] = out_shape
 
         for node in graph.get_nodes_by_types([v.op_func_name for v in NNCF_LINEAR_MODULES_DICT]):
             out_edge = graph.get_output_edges(node)[0]
```

Comment:
What if linear won't have output edge?

Reply:
Exception "out of index".
I'll add same logic for linear.

Reply:
done